packages/runner-ct/src/app/SpecList/useFuzzySort.ts

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react'
+import fuzzysort from 'fuzzysort'
+
+interface UseFuzzySort<Input, Result> {
+  search: string
+  items: Input[]
+  options: Fuzzysort.KeyOptions
+  transformResult: (spec: Input, indexes: number[]) => Result
+}
+
+export const useFuzzySort = <T, R>({ search, items, transformResult, options }: UseFuzzySort<T, R>) => {
+  const [lastSearchInput, setLastSearchInput] = React.useState<string>(search)
+  const [cachedFuzzySort, setCachedFuzzySort] = React.useState<R[]>([])
```

Comment:
super nit: I would make this naming more clear with something like `fuzzySortResults` or just `results`